src/array.rs

```diff
@@ -332,20 +347,20 @@ impl<T: Element, D: Dimension> PyArray<T, D> {
     ///     assert_eq!(arr.shape(), &[4, 5, 6]);
     /// });
     /// ```
-    pub unsafe fn new<'py, ID>(py: Python<'py>, dims: ID, is_fortran: bool) -> &Self
+    pub unsafe fn new_bound<ID>(py: Python<'_>, dims: ID, is_fortran: bool) -> Bound<'_, Self>
     where
         ID: IntoDimension<Dim = D>,
     {
         let flags = c_int::from(is_fortran);
         Self::new_uninit(py, dims, ptr::null_mut(), flags)
     }
 
-    pub(crate) unsafe fn new_uninit<'py, ID>(
-        py: Python<'py>,
+    pub(crate) unsafe fn new_uninit<ID>(
```

Comment:
I would generally prefer to kee pthe `'py` lifetime named.

Reply:
> I this case I also added #![allow(clippy::needless_lifetimes)] to silence clippy about it.

You should not need to. We manage Clippy configuration via [our `x.py` helper script](https://github.com/PyO3/rust-numpy/blob/main/x.py#L39).

Reply:
Ah, ok. I guess I must have run clippy normally. I removed it again